.readthedocs.yml

```diff
@@ -1,6 +1,41 @@
+# .readthedocs.yml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html
+# for details
+
+# Required
+version: 2
+
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+  builder: dirhtml
+  configuration: docs/conf.py
+  fail_on_warning: true  # https://github.com/pypa/setuptools/issues/2424
+
+# Build documentation with MkDocs
+#mkdocs:
+#  configuration: mkdocs.yml
+#  fail_on_warning: true
+
+# Optionally build your docs in additional formats
+# such as PDF and ePub
+formats:
+- epub
+- pdf
+- htmlzip
+
+submodules:
+  include: all  # []
+  exclude: []
+  recursive: true
+
+build:
+  image: latest
+
+# Optionally set the version of Python and requirements required
+# to build docs
 python:
-  version: 3
-  extra_requirements:
```

Comment:
The preferred way to indicate the docs requirements is through the `docs` extra. The main reason that `requirements.txt` is indicated is to avoid the bootstrapping race (perhaps a comment should be there to that effect). Is the `extra_requirements` still supported?

Reply:
@jaraco this option was unused in the old config. See `pip_install: false` two lines below + the actual build log: https://readthedocs.org/projects/setuptools/builds/12125093/.
It's supported to depend on extras: https://docs.readthedocs.io/en/stable/config-file/v2.html#packages. But personally, I've grown to dislike putting test/doc extras into dist metadata: docs deps are not runtime deps and pip doesn't know how to install extras without the dist itself, also semantically, these are not lib deps but app deps for the virtualenv that should have pinned constraints maintained with something like pip-tools ideally.
I've seen comments in `setup.cfg`/`requirements.txt` that the extra must be kept in sync with the txt file: my guess is that we're forced to do this to avoid the chicken/egg problem (but I haven't checked exactly): https://github.com/pypa/setuptools/commit/c1d7cc6701d33ee161d514ff61fe0e2017efee6c.

FWIW I've just replicated _exactly_ how it worked before, trying not to change the behavior if possible.